hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -142,21 +149,35 @@ public void handleRound(@NonNull final HederaState state, @NonNull final Round r
         // Keep track of whether any user transactions were handled. If so, then we will need to close the round
         // with the block record manager.
         final var userTransactionsHandled = new AtomicBoolean(false);
-        // handle each transaction in the round
-        round.forEachEventTransaction((event, platformTxn) -> {
-            try {
-                // skip system transactions
-                if (!platformTxn.isSystem()) {
-                    userTransactionsHandled.set(true);
-                    handlePlatformTransaction(state, event, platformTxn);
+
+        // handle each event in the round
+        for (final ConsensusEvent event : round) {
+            final var creator = networkInfo.nodeInfo(event.getCreatorId().id());
+            if (creator == null) {
+                // We were given an event for a node that *does not exist in the address book*. This will be logged as
+                // a warning, as this should never happen, and we will skip the event, which may well result in an ISS.
+                logger.warn("Received event from node {} which is not in the address book", event.getCreatorId());
```

Comment:
I guess this should never result in an ISS since the address book should be updated deterministically. Maybe the comment could be a little clearer:

```suggestion
                // We were given an event for a node that *does not exist in the address book*. This will be logged as
                // a warning, as this should never happen, and we will skip the event. The platform should guarantee
                // that we never receive an event that isn't associated with the address book, and every node in the
                // address book must have an account ID, since you cannot delete an account belonging to a node and
                // you cannot change the address book non-deterministically.
                logger.warn("Received event from node {} which is not in the address book", event.getCreatorId());
```